### ISSUE TITLE
Compile failed on Archlinux

### ISSUE THREAD

**Issue by keshav2609** [FreeRDP/Remmina#1012]
Hello,

I have been trying to install Remmina on archlinux and I am having a compilation error have a look

[ 86%] Built target remmina-plugin-nx
[ 88%] Built target remmina-plugin-xdmcp
[ 89%] Building C object remmina-plugins/rdp/CMakeFiles/remmina-plugin-rdp.dir/rdp_plugin.c.o
In file included from /tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:36:0:
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.h:276:2: erreur : unknown type name ‘HCLRCONV’
  HCLRCONV clrconv;
  ^~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c: Dans la fonction ‘remmina_rdp_pre_connect’:
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:428:2: attention : implicit declaration of function ‘freerdp_channels_pre_connect’ [-Wimplicit-function-declaration]
  freerdp_channels_pre_connect(instance->context->channels, instance);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:430:17: attention : implicit declaration of function ‘freerdp_clrconv_new’ [-Wimplicit-function-declaration]
  rfi->clrconv = freerdp_clrconv_new(CLRCONV_ALPHA);
                 ^~~~~~~~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:430:37: erreur : ‘CLRCONV_ALPHA’ undeclared (first use in this function)
  rfi->clrconv = freerdp_clrconv_new(CLRCONV_ALPHA);
                                     ^~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:430:37: note : each undeclared identifier is reported only once for each function it appears in
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c: Dans la fonction ‘remmina_rdp_post_connect’:
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:460:10: erreur : ‘CLRCONV_ALPHA’ undeclared (first use in this function)
  flags = CLRCONV_ALPHA;
          ^~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:464:12: erreur : ‘CLRBUF_32BPP’ undeclared (first use in this function)
   flags |= CLRBUF_32BPP;
            ^~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:478:12: erreur : ‘CLRBUF_16BPP’ undeclared (first use in this function)
   flags |= CLRBUF_16BPP;
            ^~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:484:2: erreur : too many arguments to function ‘gdi_init’
  gdi_init(instance, flags, NULL);
  ^~~~~~~~
In file included from /tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.h:44:0,
                 from /tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:36:
/usr/include/freerdp2/freerdp/gdi/gdi.h:541:18: note : declared here
 FREERDP_API BOOL gdi_init(freerdp\* instance, UINT32 format);
                  ^~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:489:10: erreur : ‘GDI_DC {alias struct _GDI_DC}’ has no member named ‘bitsPerPixel’
  rfi->hdc->bitsPerPixel = hdcBitsPerPixel;
          ^~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:490:10: erreur : ‘GDI_DC {alias struct _GDI_DC}’ has no member named ‘bytesPerPixel’
  rfi->hdc->bytesPerPixel = hdcBytesPerPixel;
          ^~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:507:2: attention : implicit declaration of function ‘freerdp_channels_post_connect’ [-Wimplicit-function-declaration]
  freerdp_channels_post_connect(instance->context->channels, instance);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c: Dans la fonction ‘remmina_rdp_init’:
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1104:37: attention : implicit declaration of function ‘freerdp_channels_new’ [-Wimplicit-function-declaration]
  rfi->instance->context->channels = freerdp_channels_new();
                                     ^~~~~~~~~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1104:35: attention : assignment makes pointer from integer without a cast [-Wint-conversion]
  rfi->instance->context->channels = freerdp_channels_new();
                                   ^
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c: Dans la fonction ‘remmina_rdp_close_connection’:
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1163:5: attention : implicit declaration of function ‘freerdp_channels_close’ [-Wimplicit-function-declaration]
     freerdp_channels_close(instance->context->channels, instance);
     ^~~~~~~~~~~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1186:3: attention : implicit declaration of function ‘freerdp_clrconv_free’ [-Wimplicit-function-declaration]
   freerdp_clrconv_free(rfi->clrconv);
   ^~~~~~~~~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1187:16: attention : assignment makes integer from pointer without a cast [-Wint-conversion]
   rfi->clrconv = NULL;
                ^
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1189:4: attention : implicit declaration of function ‘freerdp_channels_free’ [-Wimplicit-function-declaration]
    freerdp_channels_free(instance->context->channels);
    ^~~~~~~~~~~~~~~~~~~~~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c: Dans la fonction ‘remmina_rdp_get_screenshot’:
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1275:41: erreur : ‘rdpGdi {alias struct rdp_gdi}’ has no member named ‘bytesPerPixel’
   szmem = gdi->width \* gdi->height \* gdi->bytesPerPixel;
                                         ^~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1286:32: erreur : ‘GDI_DC {alias struct _GDI_DC}’ has no member named ‘bitsPerPixel’
   rpsd->bitsPerPixel = rfi->hdc->bitsPerPixel;
                                ^~
/tmp/yaourt-tmp-keshav/aur-remmina-git/src/remmina/remmina-plugins/rdp/rdp_plugin.c:1287:28: erreur : ‘rdpGdi {alias struct rdp_gdi}’ has no member named ‘bytesPerPixel’
   rpsd->bytesPerPixel = gdi->bytesPerPixel;
                            ^~
make[2]: **\* [remmina-plugins/rdp/CMakeFiles/remmina-plugin-rdp.dir/build.make:63: remmina-plugins/rdp/CMakeFiles/remmina-plugin-rdp.dir/rdp_plugin.c.o] Error 1
make[1]: **\* [CMakeFiles/Makefile2:454: remmina-plugins/rdp/CMakeFiles/remmina-plugin-rdp.dir/all] Error 2
make: **\* [Makefile:128: all] Error 2

Can you help please.

Thanks,

Keshav
##

**Comment by keshav2609** [comment]
anyone ? am kind of stuck!

**Comment by antenore** [comment]
We are working on it. It's an update in the FreeRDP library. https://github.com/FreeRDP/FreeRDP/issues/3530

You have to install an older version of FreeRDP (before 6 October) or wait that we will fix Remmia in the next days


@giox069

**Comment by antenore** [comment]
Maybe you can install the normal FreeRDP provided by arch and the remmina-next in AUR?

**Comment by keshav2609** [comment]
Hello,
Thanks for your answer but there is no remmina-next  in AUR...

**Comment by antenore** [comment]
https://aur.archlinux.org/packages/remmina-git

Remmina-git sorry, I've asked to the maintainer if he can help us, maybe you can join me here?

https://aur.archlinux.org/packages/remmina-git/#news

Thanks in advance for your support and patience

**Comment by giox069** [comment]
the libfreerdp_updates branch has been merged into "next".
Now remmina default branch (next) compiles with the latest FreeRDP (master) as expected.
Can we close this issue?

**Comment by lnicola** [comment]
I can confirm that this works for me now.

**Comment by giox069** [comment]
Thank you
